bridges/snowbridge/primitives/router/src/outbound/mod.rs

```diff
@@ -165,18 +183,33 @@ macro_rules! match_expression {
 	};
 }
 
-struct XcmConverter<'a, Call> {
+struct XcmConverter<'a, ConvertAssetId, Call> {
 	iter: Peekable<Iter<'a, Instruction<Call>>>,
-	ethereum_network: &'a NetworkId,
+	ethereum_network: NetworkId,
+	agent_id: AgentId,
+	_marker: PhantomData<ConvertAssetId>,
 }
-impl<'a, Call> XcmConverter<'a, Call> {
-	fn new(message: &'a Xcm<Call>, ethereum_network: &'a NetworkId) -> Self {
-		Self { iter: message.inner().iter().peekable(), ethereum_network }
+impl<'a, ConvertAssetId, Call> XcmConverter<'a, ConvertAssetId, Call>
+where
+	ConvertAssetId: MaybeEquivalence<TokenId, Location>,
+{
+	fn new(message: &'a Xcm<Call>, ethereum_network: NetworkId, agent_id: AgentId) -> Self {
+		Self {
+			iter: message.inner().iter().peekable(),
+			ethereum_network,
+			agent_id,
+			_marker: Default::default(),
+		}
 	}
 
-	fn convert(&mut self) -> Result<(AgentExecuteCommand, [u8; 32]), XcmConverterError> {
-		// Get withdraw/deposit and make native tokens create message.
-		let result = self.native_tokens_unlock_message()?;
+	fn convert(&mut self) -> Result<(Command, [u8; 32]), XcmConverterError> {
+		let result = match self.peek() {
+			Ok(ReserveAssetDeposited { .. }) => self.send_native_tokens_message(),
+			// Get withdraw/deposit and make native tokens create message.
+			Ok(WithdrawAsset { .. }) => self.send_tokens_message(),
+			Err(e) => Err(e),
+			_ => return Err(XcmConverterError::UnexpectedInstruction),
+		}?;
```

Comment:
For the benefit of reviewers, can you add a comment describing the expected form of XCM messages that are being parsed.

Example:

```
# To match transfers of Polkadot-native tokens, we expect an input of the form:

# ReserveAssetDeposited
# ClearOrigin
# BuyExecution
# ...
```

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5546/commits/f339995cdbb59419f3a0c23e7dc7df6f370aec7d